### PR TITLE
feat(routes/shu): add routes for SHU's Int'l Dept, Grad School, and Campus Highlights.

### PR DESCRIPTION
**Pull request by GhhG123**
feat(routes/shu): add routes for SHU's Int'l Dept, Grad School, and Campus Highlights.

- Corrected the root URL in `index.ts`.
- Added routes for:
  - SHU's International Department (Int'l Dept).
  - Graduate School (Grad School).
  - Campus Highlights.
- Noted the unavailability of the policy in `jwb.ts` with a comment in `index.ts`.

## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/shu/news/tzgg
/shu/news/zyxw
/shu/xykd/whxx
/shu/xykd/xsbg
/shu/jwb/tzgg
/shu/jwb/xw
/shu/gs/zhxw
/shu/gs/pygl
/shu/gs/gjjl
/shu/global/tzgg
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/shu/global.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import { Route } from '@/types';
+import cache from '@/utils/cache';
+import got from '@/utils/got';
+import { load } from 'cheerio'; // cheerio@1.0.0
+import { parseDate } from '@/utils/parse-date';
+import timezone from '@/utils/timezone';
+
+const notice_type = {
```
Comment (TonyRL) [inline lib/routes/shu/global.ts:8]: `camelCase`

File: lib/routes/shu/global.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        })
+        .get();
+
+    const items = await Promise.all(
+        list.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const detailResponse = await got(item.link); // 获取详情页内容
+                const content = load(detailResponse.data); // 使用cheerio解析内容
+
+                item.description = content('div.ej_main').html() || item.description; // 提取内容区详情
```
Comment (TonyRL) [inline lib/routes/shu/global.ts:78]: This selector matches nothing as seen in https://github.com/DIYgod/RSSHub/pull/17730#issuecomment-2503718610

File: lib/routes/shu/gs.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import { Route } from '@/types';
+import cache from '@/utils/cache';
+import got from '@/utils/got';
+import { load } from 'cheerio'; // cheerio@1.0.0
+import { parseDate } from '@/utils/parse-date';
+import timezone from '@/utils/timezone';
+
+const notice_type = {
```
Comment (TonyRL) [inline lib/routes/shu/gs.ts:8]: `camelCase`

File: lib/routes/shu/gs.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        /* headers: {
+            'user-agent': UA,
+            cookie: await getCookie(ctx),
+        }, */
+        url: notice_type[type].url,
+    });
+
+    const $ = load(response.data);
+
+    const list = $('div.only-list1 ul li') // 定位到HTML结构中的li元素
```
Comment (TonyRL) [inline lib/routes/shu/gs.ts:59]: This selector matches nothing as seen in https://github.com/DIYgod/RSSHub/pull/17730#issuecomment-2503718610

File: lib/routes/shu/gs.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        })
+        .get();
+
+    const items = await Promise.all(
+        list.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const detailResponse = await got(item.link); // 获取详情页内容
+                const content = load(detailResponse.data); // 使用cheerio解析内容
+
+                item.description = content('div.ej_main').html() || item.description; // 提取内容区详情
```
Comment (TonyRL) [inline lib/routes/shu/gs.ts:80]: This selector matches nothing

File: lib/routes/shu/index.ts [hunk 6]
```diff
@@ -1,23 +1,20 @@
...
 
-const host = 'https://www.shu.edu.cn/';
-const alias = new Map([
-    ['news', 'zhxw'], // 综合新闻
-    ['research', 'kydt1'], // 科研动态
-    ['kydt', 'kydt1'], // 科研动态
-    ['notice', 'tzgg'], // 通知公告
-    ['important', 'zyxw'], // 重要新闻
-]);
+const notice_type = {
```
Comment (TonyRL) [inline lib/routes/shu/index.ts:8]: `camelCase`

File: lib/routes/shu/index.ts [hunk 7]
```diff
@@ -28,50 +25,67 @@ export const route: Route = {
...
-            source: ['www.shu.edu.cn/:type'],
-            target: '/:type',
+            source: ['www.shu.edu.cn/'],
+            target: '/news',
         },
     ],
-    name: '官网信息',
-    maintainers: ['lonelyion'],
+    name: '官网通知公告',
+    maintainers: ['GhhG123', 'lonelyion'],
```
Comment (TonyRL) [inline lib/routes/shu/index.ts:33]:
```suggestion
    maintainers: ['lonelyion', 'GhhG123'],
```

File: lib/routes/shu/jwb.ts [hunk 8]
```diff
@@ -8,21 +8,21 @@ const host = 'https://jwb.shu.edu.cn/';
...
-            source: ['www.shu.edu.cn/:type'],
-            target: '/:type',
+            source: ['www.shu.edu.cn/index'],
+            target: '/:type?',
         },
     ],
-    name: 'Unknown',
-    maintainers: [],
+    name: '教务部',
+    maintainers: ['GhhG123'],
```
Comment (TonyRL) [inline lib/routes/shu/jwb.ts:23]:
```suggestion
    maintainers: ['tuxinghuan', 'GhhG123'],
```

File: lib/routes/shu/xykd.ts [hunk 9]
```diff
@@ -0,0 +1,95 @@
+import { Route } from '@/types';
+import cache from '@/utils/cache';
+import got from '@/utils/got';
+import { load } from 'cheerio'; // cheerio@1.0.0
+import { parseDate } from '@/utils/parse-date';
+import timezone from '@/utils/timezone';
+
+const notice_type = {
```
Comment (TonyRL) [inline lib/routes/shu/xykd.ts:8]: `camelCase`

File: lib/routes/shu/xykd.ts [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['www.shu.edu.cn/'],
+            target: '/xykd',
+        },
+    ],
+    name: '官网的校园看点',
```
Comment (TonyRL) [inline lib/routes/shu/xykd.ts:32]:
```suggestion
    name: '校园看点',
```

File: lib/routes/shu/global.ts [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+    // 发起 HTTP GET 请求
+    const response = await got({
+        method: 'get',
+
+        /* headers: {
+            'user-agent': UA,
+            cookie: await getCookie(ctx),
+        }, */
+        url: notice_type[type].url,
+    });
```
Comment (TonyRL) [inline lib/routes/shu/global.ts:53]: Use either https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/global.ts#L45-L53 or https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/global.ts#L75 but not both style

File: lib/routes/shu/gs.ts [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+  | -------- | --------- | --------- |
+  | zhxw     | pygl      | gjjl      |`,
+};
+
+async function handler(ctx) {
+    const type = ctx.req.param('type') ?? 'zhxw';
+    const rootUrl = 'https://gs.shu.edu.cn';
+
+    // 发起 HTTP GET 请求
+    const response = await got({
```
Comment (TonyRL) [inline lib/routes/shu/gs.ts:47]: Use either https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/gs.ts#L47-L55 or https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/gs.ts#L77 but not both style

File: lib/routes/shu/index.ts [hunk 13]
```diff
@@ -28,50 +25,67 @@ export const route: Route = {
...
-        .map((ele) => ({
-            title: $(ele).find('.bt').text(),
-            link: new URL($(ele).find('a').attr('href'), host).href,
-            date: $(ele).find('.sj').text(),
-        }));
+    const type = ctx.req.param('type') ?? 'tzgg';
+    const rootUrl = 'https://www.shu.edu.cn';
+
+    // 发起 HTTP GET 请求
+    const response = await got({
```
Comment (TonyRL) [inline lib/routes/shu/index.ts:46]: Use https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/index.ts#L46-L54 or https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/index.ts#L75 but not both style

File: lib/routes/shu/xykd.ts [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+  | -------- | --------- |
+  | whxx     | xsbg      |`,
+};
+
+async function handler(ctx) {
+    const type = ctx.req.param('type') ?? 'xsbg';
+    const rootUrl = 'https://www.shu.edu.cn';
+
+    // 发起 HTTP GET 请求
+    const response = await got({
```
Comment (TonyRL) [inline lib/routes/shu/xykd.ts:46]: Use either https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/xykd.ts#L46-L54 or https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/xykd.ts#L79 but not both style

File: lib/routes/shu/xykd.ts [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+                pubDate: timezone(parseDate(`${dateParts[1]}-${dateParts[0]}`, 'MM-DD'), +8), // 拼接并解析日期
+                description: item.find('div.zy').text().trim(), // 提取简要描述
+            };
+        })
+        .get();
+
+    const items = await Promise.all(
+        list.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const detailResponse = await got(item.link); // 获取详情页内容
```
Comment (TonyRL) [inline lib/routes/shu/xykd.ts:79]: Use either https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/xykd.ts#L46-L54 or https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/xykd.ts#L79 but not both style

File: lib/routes/shu/index.ts [hunk 16]
```diff
@@ -28,50 +25,67 @@ export const route: Route = {
...
+
+    const items = await Promise.all(
         list.map((item) =>
             cache.tryGet(item.link, async () => {
-                const response = await got.get(item.link);
-                const $ = load(response.data);
-                item.author = $('.xx>:nth-child(2)').text().trim().slice(3); // 投稿：xxx
-                item.pubDate = parseDate(item.date, 'YYYY.MM.DD');
-                item.description = $('.v_news_content').html() || item.title;
+                const detailResponse = await got(item.link);
```
Comment (TonyRL) [inline lib/routes/shu/index.ts:75]: Use https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/index.ts#L46-L54 or https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/index.ts#L75 but not both style

File: lib/routes/shu/gs.ts [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+                pubDate: timezone(parseDate(pubDate, 'YYYY年MM月DD日'), +8), // 解析并转换日期
+                description: '', // 没有提供简要描述，设为空字符串
+            };
+        })
+        .get();
+
+    const items = await Promise.all(
+        list.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const detailResponse = await got(item.link); // 获取详情页内容
```
Comment (TonyRL) [inline lib/routes/shu/gs.ts:77]: Use either https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/gs.ts#L47-L55 or https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/gs.ts#L77 but not both style

File: lib/routes/shu/global.ts [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+                pubDate: timezone(parseDate(pubDate, 'YYYY年MM月DD日'), +8), // 解析并转换日期
+                description: '', // 没有提供简要描述，设为空字符串
+            };
+        })
+        .get();
+
+    const items = await Promise.all(
+        list.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const detailResponse = await got(item.link); // 获取详情页内容
```
Comment (TonyRL) [inline lib/routes/shu/global.ts:75]: Use either https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/global.ts#L45-L53 or https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/global.ts#L75 but not both style

File: lib/routes/shu/global.ts (hunk 2 above)
Comment (GhhG123) [reply lib/routes/shu/global.ts]: I understand, and have fixed all the same errors.

**Review by GhhG123 — COMMENTED** [review]
Thank you for your patience in raising my question. This is my first PR.😊

**Pull request merged**